pandas/core/dtypes/dtypes.py

```diff
@@ -243,8 +259,12 @@ def _from_categorical_dtype(cls, dtype, categories=None, ordered=None):
         return cls(categories, ordered)
 
     @classmethod
-    def _from_values_or_dtype(cls, values=None, categories=None, ordered=None,
-                              dtype=None):
+    def _from_values_or_dtype(cls,
+                              values=None,
```

Comment:
Can you annotate values / categories?

Reply:
See comment above. The most precise here would again be ``Iterable``, but same problems as above.